src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1074,13 +1118,10 @@ namespace winrt::TerminalApp::implementation
         // current list of tabs.
         const auto tabSwitchMode = _settings.GlobalSettings().TabSwitcherMode();
         const bool commandPaletteIsVisible = CommandPalette().Visibility() == Visibility::Visible;
-        if (tabSwitchMode == TabSwitcherMode::MostRecentlyUsed && commandPaletteIsVisible)
+        if (commandPaletteIsVisible)
         {
-            CommandPalette().SetTabActions(_mruTabActions, false);
-        }
-        else if (commandPaletteIsVisible)
-        {
-            _UpdatePaletteWithInOrderTabs();
+            auto tabs = tabSwitchMode == TabSwitcherMode::MostRecentlyUsed ? _mruTabs : _tabs;
```

Comment:
just to spare a ref count copy... you might be able to use `auto& tabs`

```suggestion
            auto& tabs = tabSwitchMode == TabSwitcherMode::MostRecentlyUsed ? _mruTabs : _tabs;
```